File: part2-nosql/mongodb_operations.py
from pymongo import MongoClient

def main():
    client = MongoClient("mongodb://localhost:27017/")
    db = client.fleximart_nosql
    collection = db.products

    # Insert sample documents
    collection.insert_many([
        {"product_id": "P006", "name": "Organic Honey", "category": "Groceries", "price": 450},
        {"product_id": "P007", "name": "HP Laptop", "category": "Electronics", "price": 52999}
    ])

    # Find all products
    for product in collection.find():
        print(product)

    # Aggregation: count products per category
    pipeline = [
        {"$group": {"_id": "$category", "count": {"$sum": 1}}}
    ]
    result = collection.aggregate(pipeline)
    for r in result:
        print(r)

if __name__ == "__main__":
    main()

File: part2-nosql/mongodb_operations.js
use fleximart_nosql;

// Load data
db.products.insertMany([
  {
    product_id: "ELEC001",
    name: "Samsung Galaxy S21",
    category: "Electronics",
    price: 79999,
    stock: 150,
    reviews: [{user:"U001",rating:5,comment:"Great",date:new Date("2024-01-15")}]
  }
]);

// Query 1
db.products.find(
  {category:"Electronics", price:{$lt:50000}},
  {name:1,price:1,stock:1,_id:0}
);

// Query 2
db.products.aggregate([
  {$unwind:"$reviews"},
  {$group:{_id:"$name",avg_rating:{$avg:"$reviews.rating"}}},
  {$match:{avg_rating:{$gte:4}}}
]);

// Update
db.products.updateOne(
  {product_id:"ELEC001"},
  {$push:{reviews:{user:"U999",rating:4,comment:"Good value",date:new Date()}}}
);

// Aggregation
db.products.aggregate([
  {$group:{_id:"$category",avg_price:{$avg:"$price"},product_count:{$sum:1}}},
  {$sort:{avg_price:-1}}
]);
